x/emissions/migrations/v4/migrate.go

```diff
@@ -156,6 +174,61 @@ func MigrateTopics(
 	return nil
 }
 
+// Deletes all keys in the store with the given keyPrefix `maxPageSize` keys at a time
+func safelyClearWholeMap(store storetypes.KVStore, keyPrefix []byte) {
+	s := prefix.NewStore(store, keyPrefix)
+
+	// Loop until all keys are deleted.
+	// Unbounded not best practice but we are sure that the number of keys will be limited
+	// and not deleting all keys means "poison" will remain in the store.
+	for {
+		// Gather keys to eventually delete
+		iterator := s.Iterator(nil, nil)
+		keysToDelete := make([][]byte, 0)
+		count := uint64(0)
+		for ; iterator.Valid(); iterator.Next() {
+			if count >= maxPageSize {
+				break
+			}
+
+			keysToDelete = append(keysToDelete, iterator.Key())
+			count++
+		}
+		iterator.Close()
+
+		// If no keys to delete, break => Exit whole function
+		if len(keysToDelete) == 0 {
+			break
+		}
+
+		// Delete the keys
+		for _, key := range keysToDelete {
+			s.Delete(key)
+		}
+	}
+}
+
+// Clear out poison NaN values on different inferences, scores etc
+func ResetMapsWithNonNumericValues(store storetypes.KVStore, cdc codec.BinaryCodec) {
+	safelyClearWholeMap(store, emissionstypes.InferenceScoresKey)
+	safelyClearWholeMap(store, emissionstypes.ForecastScoresKey)
+	safelyClearWholeMap(store, emissionstypes.ReputerScoresKey)
+	safelyClearWholeMap(store, emissionstypes.InfererScoreEmasKey)
+	safelyClearWholeMap(store, emissionstypes.ForecasterScoreEmasKey)
+	safelyClearWholeMap(store, emissionstypes.ReputerScoreEmasKey)
+	safelyClearWholeMap(store, emissionstypes.AllLossBundlesKey)
+	safelyClearWholeMap(store, emissionstypes.NetworkLossBundlesKey)
+	safelyClearWholeMap(store, emissionstypes.InfererNetworkRegretsKey)
+	safelyClearWholeMap(store, emissionstypes.ForecasterNetworkRegretsKey)
+	safelyClearWholeMap(store, emissionstypes.OneInForecasterNetworkRegretsKey)
+	safelyClearWholeMap(store, emissionstypes.LatestNaiveInfererNetworkRegretsKey)
+	safelyClearWholeMap(store, emissionstypes.LatestOneOutInfererInfererNetworkRegretsKey)
+	safelyClearWholeMap(store, emissionstypes.LatestOneOutInfererForecasterNetworkRegretsKey)
+	safelyClearWholeMap(store, emissionstypes.LatestOneOutForecasterInfererNetworkRegretsKey)
+	safelyClearWholeMap(store, emissionstypes.LatestOneOutForecasterForecasterNetworkRegretsKey)
```

Comment:
Please add logs to each line so we're able to track progress this time around.

Reply:
addressed in [b9ade84](https://github.com/allora-network/allora-chain/pull/590/commits/b9ade8439d6adf7fa88f50b44cf72cf6bb502aa7)